Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -87,6 +93,7 @@ private void SaveSettings(DiagnosticsSettings diagnosticsSettings)
                 FileSystemHelpers.EnsureDirectory(Path.GetDirectoryName(_path));
             }
 
+            diagnosticsSettings.EnsureValid();
             string fileContent = JsonConvert.SerializeObject(diagnosticsSettings);
```

Comment:
this is to prevent rubbish value, where user can call UpdateSetting("AzureDriveTraceLevel", 9999)

since we are doing validation, i think will be good to also prevent this kind of invalid value

````
           var result = (TraceEventType)2;
            Console.WriteLine(result);                         // will print Error

            var result2 = (TraceEventType)9999;
            Console.WriteLine(result2);                       // will print 9999
````